internal/vulnstore/postgres/putvulnerabilities.go

```diff
@@ -113,11 +117,13 @@ func putVulnerabilities(ctx context.Context, pool *pgxpool.Pool, updater string,
 	}
 	defer tx.Rollback(ctx)
 
+	skipCt := 0
 	// safe sized batch inserts to postgres
 	mBatcher := microbatch.NewInsert(tx, 2000, time.Minute)
 	for _, vuln := range vulns {
-		if vuln.Package == nil {
-			vuln.Package = &claircore.Package{}
+		if vuln.Package == nil || vuln.Package.Name == "" {
+			skipCt++
+			continue
 		}
 		if vuln.Dist == nil {
```

Comment:
@hdonnay 
I'm thinking... if we do not have a distribution associated with the vulnerability - we don't match it either...
With your knowledge of the RPM stuff as well does it make sense to ignore vulns where we can't get a distribution either? Most of the time we munge this field ourselves anyway.

Reply:
I'm fine with that, I think. I'll think about it a bit and complain if I can come up with a counterexample.

Reply:
@hdonnay 

Thinking about it... all matchers return a MatchContraint which means they require their pending vulns to have dists as well.